forge-game/pom.xml

```diff
@@ -34,5 +35,10 @@
             <artifactId>sentry-logback</artifactId>
             <version>7.15.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.jgrapht</groupId>
```

Comment:
guava has its own graph lib you might want to check out?

Reply:
I compared them but it's rather basic...

Most importantly this already gives us cycle finding methods which if you check out those classes isn't all that trivial
I really do not want to reimplement one of these by hand 😅
E.g. here's the paper for the one I've chosen: https://link.springer.com/article/10.1007/BF01931370

Second bonus is that it also has support for visualization which I may try to add at some point too :)